src/FSharp.FGL.IO/GDF.fs

```diff
@@ -136,3 +136,84 @@ module GDF =
+        else x
+
+    //Reconstructs the header of an edge or node based on a Map<string,GDFValue> 
+    let private reconstructHeader (value:Map<string,GDFValue>) =
+        let headerString    =   value|>Map.toList|>List.map(fst)
+        let headerTypes     =   value|>getGDFValue 
+        let header          =   List.map2 (fun a b -> (a)+" "+(b)) headerString headerTypes
+        let defIndex        =   header|>List.findIndex(fun x -> x.Contains "def>")
+        let headerSortedNew =   ((header.[defIndex])::header)|>List.distinct
+        String.concat ";" headerSortedNew
+
+    //Reconstructs vertices or edges based on LVertex or LEdge
+    let private reconstructVertices (value:LVertex<GDFValue,Map<string,GDFValue>>)     =
+        let index               =   ("nodedef>name",(fst value))        
+        let valueList           =   index::(snd value|>Map.toList)|>List.distinct
+        let updatedValueList    =   valueList|>List.map(snd)|>List.map(fun x -> gdfValueToString x)
+        let nameIndex           =   valueList|>List.map(fst)|>List.findIndex (fun x -> x.Contains "def>")
+        let valueListSortedNew  =   ((updatedValueList.[nameIndex])::updatedValueList)|>List.distinct|> List.map(addQuotationsIfNeeded)
+        String.concat ";" valueListSortedNew
+    let private reconstructEdges (value:LEdge<GDFValue,Map<string,GDFValue>>)     =
+        let index1              =   ("edgedef>node1"),((fun (a,b,c) -> a) value)
+        let index2              =   ("node2"),((fun (a,b,c) -> b) value)
+        let valueList           =   index1::index2::((fun (a,b,c) -> c) value|>Map.toList)|>List.distinct
+        let updatedValueList    =   valueList|>List.map(snd)|>List.map(fun x -> gdfValueToString x)
+        let nameIndex           =   valueList|>List.map(fst)|>List.findIndex (fun x -> x.Contains "def>")
+        let valueListSortedNew  =   ((updatedValueList.[nameIndex])::updatedValueList)|>List.distinct|> List.map(addQuotationsIfNeeded)
+        String.concat ";" valueListSortedNew
+
+    ////WIP, searches for values, that are described in the header but are not present in the value list and adds them as an empty string
+    //let addEmptyValuesIfNeeded (value:(string*GDFValue) list) (header:string)=
+    //    let headerList :string list     =   header.Split ";"|>List.ofArray|>List.sort
+    //    let valueGDFTypes :string list  =   List.map fst value|>List.sort
+    //    if headerList = valueGDFTypes then
+    //        value
+    //    else
+    //       ((),())::value|>List.sort
+       
+    //Constructs an Array containing vertex header,vertices,edge header and edges based on an LVertex list and an LEdge list
+    let private vertexListEdgeListToStringArray (vertexList:LVertex<GDFValue,Map<string,GDFValue>> list) (edgeList:LEdge<GDFValue,Map<string,GDFValue>> list) : string[]=
+        let vertexHeader    = List.map snd vertexList|>List.map reconstructHeader|>List.distinct|>String.concat ";"
```

Comment:
The way the distinction is implemented here will not achieve what you aimed for, as the items of the headers of each vertex are merged separately. E.g:
- One vertex has `id,label,color`
- The other vertex has `id,label`
The full strings will be checked as a whole for distinctness and then merged, resulting in `id,label,color,id,label`.

You need to first pool the items of all headers of all vertices, choose the distinct entries and then merge